mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td

```diff
@@ -186,6 +209,29 @@ def Mesh_ShardOp : Mesh_Op<"shard", [Pure, SameOperandsAndResultType]> {
   }];
 }
 
+def Mesh_ProcessIndexOp : Mesh_Op<"process_index", [Pure, DeclareOpInterfaceMethods<SymbolUserOpInterface>]> {
```

Comment:
Is there an Int value representing the device/process id, not a list of int?

Reply:
I assume you mean to have the correspondence between multi index and flat/linear index.
Do you want such an operation explicitly in the IR that extracts the linear index?
If it is to be used during lowering we can have a C++ function that adds IR to flatten the index. There may be already somewhere in MLIR such a thing.

Reply:
Yeah, a flat index is what I mean. I think both an explicit IR and implementation during lowering is fine. 

But we'd better clearly wrote the doc clearly how the multi-index is converted to flat index.

E.g. a mesh cluster of 2x2, (1, 0) represents device id 1 or 2 might need to be defined.

We can leave this doc to the next PR.